### PR TITLE
{Relay} `az relay`: Migrate to atomic commands

### PR DESCRIPTION
**Pull request by ZengTaoxu**
**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/relay/custom.py [hunk 1]
```diff
@@ -2,136 +2,86 @@
...
+class WcfrelayUpdate(_WcfrelayUpdate):
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        from azure.cli.core.aaz import AAZStrArg
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.status = AAZStrArg(
+            options=["--status"],
+            help="Enumerates the possible values for the status of a messaging entity.",
+            enum={"Active": "Active", "Disabled": "Disabled", "ReceiveDisabled": "ReceiveDisabled", "SendDisabled": "SendDisabled"},
+        )
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/relay/custom.py:73]: Missing `nullable=True`?

File: src/azure-cli/azure/cli/command_modules/relay/custom.py [hunk 2]
```diff
@@ -2,136 +2,86 @@
...
+class HycoUpdate(_HycoUpdate):
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        from azure.cli.core.aaz import AAZStrArg
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.status = AAZStrArg(
+            options=["--status"],
+            help="Enumerates the possible values for the status of a messaging entity.",
+            enum={"Active": "Active", "Disabled": "Disabled", "ReceiveDisabled": "ReceiveDisabled", "SendDisabled": "SendDisabled"},
+        )
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/relay/custom.py:86]: Same as above.

**Comment by bebound** [comment]
Please also remove `azure-mgmt-relay` from dependency.

File: src/azure-cli/azure/cli/command_modules/relay/_client_factory.py [hunk 3]
```diff
@@ -14,9 +14,5 @@ def namespaces_mgmt_client_factory(cli_ctx, _):
     return cf_relay(cli_ctx).namespaces
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/relay/_client_factory.py:14]: Please clean up the code in client_factory

**Review by necusjz — APPROVED** [review]
LGTM

**Review by kairu-ms — APPROVED** [review]

**Review by jsntcy — APPROVED** [review]

**Comment by jiasli** [comment]
`k8s-extension` extension relies on `azure.mgmt.relay`. 

This removal of `azure-mgmt-relay` causes CI failure:

https://dev.azure.com/azclitools/public/_build/results?buildId=87056&view=logs&j=7621d84f-e7df-5fb0-9a4f-5fbdcb3421ee&t=bbcd1db7-629e-5645-8ed8-a18b90e0a11a&l=2817

```
  File "/opt/az/azcliextensions/k8s-extension/azext_k8s_extension/custom.py", line 30, in <module>
    from .partner_extensions.AzureMLKubernetes import AzureMLKubernetes
  File "/opt/az/azcliextensions/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py", line 15, in <module>
    import azure.mgmt.relay
ModuleNotFoundError: No module named 'azure.mgmt.relay'
```

(The test for `k8s-extension` is temporarily disabled by https://github.com/Azure/azure-cli/pull/27285.)

**Pull request merged**